pkgpanda/util.py

```diff
@@ -150,6 +151,38 @@ def get_requests_retry_session(max_retries=4, backoff_factor=1, status_forcelist
     return session
 
 
+def stream_remote_file_with_retries(out_filename, url, retries=4):
+    def cleanup():
+        try:
+            os.remove(out_filename)
+        except Exception:
+            pass
+
+    while True:
+        with open(out_filename, "w+b") as f:
+            r = get_requests_retry_session().get(url, stream=True)
```

Comment:
By default, `get` will follow redirects (http://docs.python-requests.org/en/v3.0.0/_modules/requests/api/#get). If you really want to ensure that the request did not get redirected you should pass `allow_redirects=False` here.